alcs-frontend/src/app/services/notice-of-intent/decision/notice-of-intent-decision.service.ts

```diff
@@ -0,0 +1,119 @@
+import { ToastService } from '../../toast/toast.service';
+import {
+  NoticeOfIntentDecisionDto,
+  CreateNoticeOfIntentDecisionDto,
+  NoticeOfIntentDecisionOutcomeCodeDto,
+  UpdateNoticeOfIntentDecisionDto,
+} from './notice-of-intent-decision.dto';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NoticeOfIntentDecisionService {
+  private url = `${environment.apiUrl}/notice-of-intent-decision`;
+
+  constructor(private http: HttpClient, private toastService: ToastService) {}
+
+  async fetchByFileNumber(fileNumber: string) {
+    let decisions: NoticeOfIntentDecisionDto[] = [];
+    try {
+      decisions = await firstValueFrom(
+        this.http.get<NoticeOfIntentDecisionDto[]>(`${this.url}/notice-of-intent/${fileNumber}`)
+      );
+    } catch (err) {
+      this.toastService.showErrorToast('Failed to fetch decisions');
+    }
+    return decisions;
+  }
+
+  async fetchCodes() {
+    let outcomes: NoticeOfIntentDecisionOutcomeCodeDto[] = [];
+    try {
+      const res = await firstValueFrom(
+        this.http.get<{
+          outcomes: NoticeOfIntentDecisionOutcomeCodeDto[];
+        }>(`${this.url}/codes`)
+      );
+      outcomes = res.outcomes;
+    } catch (err) {
+      this.toastService.showErrorToast('Failed to fetch decisions');
+    }
+    return {
+      outcomes,
+    };
+  }
+
+  async update(uuid: string, data: UpdateNoticeOfIntentDecisionDto) {
+    try {
+      const res = await firstValueFrom(this.http.patch<NoticeOfIntentDecisionDto>(`${this.url}/${uuid}`, data));
+      this.toastService.showSuccessToast('Decision updated');
+      return res;
+    } catch (e) {
+      if (e instanceof HttpErrorResponse && e.status === 400 && e.error?.message) {
+        this.toastService.showErrorToast(e.error.message);
+      } else {
+        this.toastService.showErrorToast('Failed to update decision');
+      }
+      throw e;
+    }
+  }
+
+  async create(decision: CreateNoticeOfIntentDecisionDto) {
+    try {
+      const res = await firstValueFrom(this.http.post<NoticeOfIntentDecisionDto>(`${this.url}`, decision));
+      this.toastService.showSuccessToast('Decision created');
+      return res;
+    } catch (e) {
+      if (e instanceof HttpErrorResponse && e.status === 400 && e.error?.message) {
+        this.toastService.showErrorToast(e.error.message);
+      } else {
+        this.toastService.showErrorToast(`Failed to create decision`);
+      }
+      throw e;
+    }
+  }
+
+  async delete(uuid: string) {
+    try {
+      await firstValueFrom(this.http.delete<NoticeOfIntentDecisionDto>(`${this.url}/${uuid}`));
+      this.toastService.showSuccessToast('Decision deleted');
+    } catch (err) {
+      this.toastService.showErrorToast('Failed to delete decision');
+    }
+  }
+
+  async uploadFile(decisionUuid: string, file: File) {
+    if (file.size > environment.maxFileSize) {
+      const niceSize = environment.maxFileSize / 1048576;
```

Comment:
I guess it is good time to move this '1048576' into some kind of constant

Reply:
Created a file size checker that has a clearly defined constant and replaced all usages (other than Portal) with that